solidity/test/isms/CCIPIsm.t.sol

```diff
@@ -0,0 +1,364 @@
+
+        vm.selectFork(mainnetFork);
+
+        bytes memory encodedHookData = abi.encode(messageId);
+
+        l1Mailbox.updateLatestDispatchedId(messageId);
+
+        uint256 quotedFee = ccipHookMainnet.quoteDispatch(
+            testMetadata,
+            encodedMessage
+        );
+
+        ccipHookMainnet.postDispatch{value: (quotedFee * 11) / 10}(
+            testMetadata,
+            encodedMessage
+        );
+    }
+
+    function testFork_postDispatch_RevertWhen_NotEnoughValueSent() public {
+        deployAll();
+
+        vm.selectFork(mainnetFork);
+
+        l1Mailbox.updateLatestDispatchedId(messageId);
+
+        vm.expectRevert();
+
+        ccipHookMainnet.postDispatch(testMetadata, encodedMessage);
+    }
+
+    function testFork_postDispatch_RevertWhen_NotLastDispatchedMessage()
+        public
+    {
+        deployAll();
+
+        vm.selectFork(mainnetFork);
+
+        vm.expectRevert(
+            "AbstractMessageIdAuthHook: message not latest dispatched"
+        );
+
+        ccipHookMainnet.postDispatch(testMetadata, encodedMessage);
+    }
+
+    // /* ============ ISM.verifyMessageId ============ */
+
+    function testFork_verifyMessageId() public {
+        deployAll();
+
+        vm.selectFork(optimismFork);
+
+        Client.Any2EVMMessage memory message = _encodeCCIPReceiveMessage();
+
+        vm.prank(OP_ROUTER_ADDRESS);
+        ccipISMOptimism.ccipReceive(message);
+
+        assertTrue(ccipISMOptimism.verifiedMessages(messageId).isBitSet(255));
+    }
+
+    function testFork_verifyMessageId_RevertWhen_SenderNotAllowed() public {
+        deployAll();
+
+        vm.selectFork(optimismFork);
+
+        Client.Any2EVMMessage memory message = _encodeCCIPReceiveMessage();
+        message.sender[0] = ~message.sender[0];
+
+        vm.prank(OP_ROUTER_ADDRESS);
+        vm.expectRevert("Unauthorized hook");
+        ccipISMOptimism.ccipReceive(message);
+    }
+
+    function testFork_verifyMessageId_RevertWhen_SourceChainNotAllowed()
+        public
+    {
+        deployAll();
+
+        vm.selectFork(optimismFork);
+
+        Client.Any2EVMMessage memory message = _encodeCCIPReceiveMessage();
+        message.sourceChainSelector = ~message.sourceChainSelector;
+
+        vm.prank(OP_ROUTER_ADDRESS);
+        vm.expectRevert("Unauthorized origin");
+        ccipISMOptimism.ccipReceive(message);
+    }
+
+    function testFork_verifyMessageId_RevertWhen_InvalidRouter() public {
+        deployAll();
+
+        vm.selectFork(optimismFork);
+
+        vm.startPrank(MAINNET_ROUTER_ADDRESS);
+
+        Client.Any2EVMMessage memory message = _encodeCCIPReceiveMessage();
+
+        bytes4 selector = bytes4(keccak256("InvalidRouter(address)"));
+        vm.expectRevert(
+            abi.encodeWithSelector(selector, MAINNET_ROUTER_ADDRESS)
+        );
+
+        ccipISMOptimism.ccipReceive(message);
+
+        vm.stopPrank();
+    }
+
+    function testFork_preVerifyMessageId_RevertWhen_UnauthorizedRouter()
```

Comment:
Is this the test that checks that not just anybody can call preVerify? If so, can you make this a function argument again (or does that not work with forks?)